docs/plugins/discovery-ec2.asciidoc

```diff
@@ -1,167 +1,214 @@
 
-`read_timeout`::
+`discovery.ec2.read_timeout`::
 
-    The socket timeout for connecting to ec2. The value should specify the unit. For example,
-    a value of `5s` specifies a 5 second timeout. The default value is 50 seconds.
+    The socket timeout for connections to EC2,
+    {ref}/common-options.html#time-units[including the units]. For example, a
+    value of `60s` specifies a 60-second timeout. Defaults to 50 seconds.
 
-`groups`::
+`discovery.ec2.groups`::
 
-    Either a comma separated list or array based list of (security) groups.
-    Only instances with the provided security groups will be used in the
-    cluster discovery. (NOTE: You could provide either group NAME or group
-    ID.)
+    A list of the names or IDs of the security groups to use for discovery. The
+    `discovery.ec2.any_group` setting determines the behaviour of this setting.
+    Defaults to an empty list, meaning that security group membership is
+    ignored by EC2 discovery.
 
-`host_type`::
+`discovery.ec2.any_group`::
+
+    Defaults to `true`, meaning that instances belonging to _any_ of the
+    security groups specified in `discovery.ec2.groups` will be used for
+    discovery. If set to `false`, only instances that belong to _all_ of the
+    security groups specified in `discovery.ec2.groups` will be used for
+    discovery.
+
+`discovery.ec2.host_type`::
 
 +
 --
-The type of host type to use to communicate with other instances. Can be
-one of `private_ip`, `public_ip`, `private_dns`, `public_dns` or `tag:TAGNAME` where
-`TAGNAME` refers to a name of a tag configured for all EC2 instances. Instances which don't
-have this tag set will be ignored by the discovery process.
 
-For example if you defined a tag `my-elasticsearch-host` in ec2 and set it to `myhostname1.mydomain.com`, then
-setting `host_type: tag:my-elasticsearch-host` will tell Discovery Ec2 plugin to read the host name from the
-`my-elasticsearch-host` tag. In this case, it will be resolved to `myhostname1.mydomain.com`.
-http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html[Read more about EC2 Tags].
+Each EC2 instance has a number of different addresses that might be suitable
+for discovery. This setting allows you to select which of these addresses is
+used by the discovery process. It can be set to one of `private_ip`,
+`public_ip`, `private_dns`, `public_dns` or `tag:TAGNAME` where `TAGNAME`
+refers to a name of a tag. This setting defaults to `private_ip`.
 
-Defaults to `private_ip`.
---
+If you set `discovery.ec2.host_type` to a value of the form `tag:TAGNAME` then
+the value of the tag `TAGNAME` attached to each instance will be used as that
+instance's address for discovery. Instances which do not have this tag set will
+be ignored by the discovery process.
 
-`availability_zones`::
+For example if you tag some EC2 instances with a tag called
+`my-elasticsearch-host` and set `host_type: tag:my-elasticsearch-host` then the
```

Comment:
nit: would be good to use a tag-name in the example that is production friendly, such as `elasticsearch-host-name`.  I think some would use the names from this example in production.

Reply:
Good point, thanks. I adjusted it in e9853c709682cfeb19b48c935a4f21cbfd69cbd5.